lambdaclass/curse_of_myrra
# Setup player abilities dynamically

Since ScripteableObject lives in the editor, we couldnt change them values ar run-time. So the solution is to Instanciate a copy of them and use it.
With this PR we can instanciate a copy of the skills ScripteableObject maping the backend data from the Skills.json. 

It also destroy the instances when the player is dead. 

SO stands for ScripteableObject

How it works?
- We have the original Skills SO stored in a list inside the Character SO
- In the `CustomLevelManager` we iterate over them and create a clone of each one.
- Then we map the Skill.json data to the copies using the constructor `skillClone.InitWithBackend()`
- Then we assing them to the Skills
- Finally when the player is dead we `Destroy` each clone instance.

To test:
- Play a Game for example with H4ck
- Use the multishot skill and see the angle of the cone
- Exit the game ( but dont stop the playmode)
- Edit the Skill.json for the Multiushot skill. Increasing the angle.
- Start a new game again and check if the angle has changed.
